homeassistant/components/roborock/coordinator.py

```diff
@@ -178,7 +179,13 @@ def parse_image(self, map_bytes: bytes) -> bytes | None:
             return None
         img_byte_arr = io.BytesIO()
         parsed_map.image.data.save(img_byte_arr, format=MAP_FILE_FORMAT)
-        return img_byte_arr.getvalue()
+        image = img_byte_arr.getvalue()
+        if self.current_map is not None and image != self.maps[self.current_map].image:
```

Comment:
This still has the side effect of doing persistence that I mentioned which I still think is surprising. Its not parsing, but also saving. I think just move this out to the caller.